packages/default/scss/drawer/_layout.scss

```diff
@@ -0,0 +1,168 @@
+
+            ul {
+                padding: 0;
+                margin: 0;
+            }
+
+            .k-drawer-item {
+                display: block;
+                line-height: $drawer-item-line-height;
+                font-size: $drawer-item-font-size;
+                padding: $drawer-item-padding-y $drawer-item-padding-x;
+                list-style: none;
+
+                &.k-drawer-separator {
+                    padding: 0;
+                    height: 1px;
+                }
+            }
+
+            .k-icon {
+                display: inline-block;
+                font-size: $drawer-icon-size;
+                padding: 0 $drawer-item-padding-x;
+            }
+
+            .k-item-text {
+                display: inline-block;
+                padding: 0 $drawer-item-padding-x;
+                vertical-align: middle;
+            }
+        }
+
+        &.expanded {
+            border-right-width: $drawer-border-width;
+
+            .k-drawer-items {
+                width: $drawer-initial-width; /* Default width of sidebar */
+            }
+        }
+    }
+
+    .k-drawer-wrapper {
+        overflow-x: hidden;
+
+        &.mini-mode {
+            .k-drawer-container {
+                border-right-width: $drawer-border-width;
+            }
+
+            .k-drawer-items {
+                overflow-y: visible;
+                overflow-x: initial;
+                width: $drawer-mini-initial-width; /* Default width of mini sidebar */
+            }
+
+             &.expanded .k-drawer-items {
+                width: $drawer-initial-width; /* Default width of sidebar */
+            }
+        }
+
+        &.overlay {
+            .k-drawer-container {
+                position: fixed;
+                left: 0;
+                right: auto;
+                height: 100%;
+                z-index: 10002;
+            }
+
+            .k-drawer-items {
+                max-width: 80vw;
+            }
+        }
+
+
+        &.push {
+            display: flex;
+
+            .k-drawer-content {
+                flex: 1 1 auto;
+            }
+
+            .k-drawer-container {
+                position: relative;
+                flex: 0 0 auto;
+            }
+        }
+
+        .k-overlay {
+            display: none;
+        }
+
+        &.expanded  {
+            .k-overlay {
+                display: block;
+            }
+        }
+
+        &.right {
+            &.overlay {
+                .k-drawer-container {
+                    left: auto;
+                    right: 0;
+                }
+            }
+
+            &.push {
+                .k-drawer-container {
+                    order: 2;
+                }
+            }
+
+            .k-drawer-container.expanded,
+            &.mini-mode .k-drawer-container {
+                border-left-width: $drawer-border-width;
+                border-right-width: 0;
+            }
+        }
+    }
+
+    .k-rtl,
+    [dir="rtl"] {
```

Comment:
This selector will not work if the `dir="rtl"` attribute is assigned to the `k-drawer` element;

The selector should look like:

```
.k-rtl .k-drawer,
.k-drawer[dir="rtl"],
[dir="rtl"] .k-drawer{
}


Also, the below code is not handling the `overlay` scenario; are we going to support this?
```

Reply:
the overlay mode adds fixed position and left or right 0 depending on the position so we don't need additional styling in the RTL